source/iostreams.tex

```diff
@@ -5044,35 +5007,28 @@
 the function endeavors
 to obtain the requested input.
 Otherwise, if the \tcode{sentry} constructor exits by throwing an exception or if
-the \tcode{sentry} object returns \tcode{false}, when converted to a value of type
+the \tcode{sentry} object produces \tcode{false}, when converted to a value of type
 \tcode{bool},
 the function returns without attempting to obtain any input.
 In either case the number of extracted characters is set to 0;
 unformatted input functions taking a character array of nonzero size as
 an argument shall also store a null character (using
 \tcode{charT()})
 in the first location of the array.
+If \tcode{rdbuf()->sbumpc()} or \tcode{rdbuf()->sgetc()}
+return \tcode{traits::eof()}, then
+\tcode{ios_base::eofbit} is set in the local error state and
+the input function stops trying to obtain the requested input.
 If an exception is thrown during input then
-\tcode{ios_base::badbit}
-is turned on
-\begin{footnote}
-This is done without causing an
-\tcode{ios_base::failure}
-to be thrown.
-\end{footnote}
-in
-\tcode{*this}'s
-error state.
-(Exceptions thrown from
-\tcode{basic_ios<>::clear()}
-are not caught or rethrown.)
-If
-\tcode{(exceptions()\&badbit) != 0}
-then the exception is rethrown.
-It also counts the number of characters extracted.
-If no exception has been thrown it ends
-by storing the count in a member object
-and returning the value specified.
+\tcode{ios_base::badbit} is set in the local error state,
+\tcode{*this}'s error state is set to the local error state, and
+the exception is rethrown if \tcode{(exceptions() \& badbit) != 0}.
```

Comment:
Another pre-existing use of `badbit` instead of `ios_base::badbit`.

Reply:
#6019